packages/core/src/hooks/useHealthFactor.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react'
+import { useQuery } from '@apollo/client'
+
+import { bigNumberToNumber } from '../utilities'
+import { HealthFactorQuery } from '../graphql'
+
+import {
+  GET_HEALTH_FACTOR_DATA,
+  HealthFactorGraph,
+  UserReserveVariables,
+} from '@bloxifi/core'
+
+/**
+ * Options that can be used to configure useHealthFactor() hook.
+ */
+export interface Props {
+  /**
+   * User wallet account
+   */
+  readonly currentAccount?: string
+}
+
+/**
+ * Hook that returns calculated Health Factor value
+ */
+export const useHealthFactor = ({ currentAccount }: Props = {}): number => {
+  const [value, setValue] = useState(undefined)
+
+  const { data } = useQuery<HealthFactorGraph, UserReserveVariables>(
+    GET_HEALTH_FACTOR_DATA,
+    {
+      variables: {
+        user: currentAccount?.toLowerCase(),
+      },
+      onCompleted: () => calculate(),
+    },
+  )
+
+  const calculateTotalBorrow = (array: HealthFactorQuery[]): number =>
```

Comment:
we can move `calculateTotalBorrow` and `calculateTotalCollateralWithLT` outside this hook, it should be in this file, but not in the hook, since they are independent functions.